WordPressCom-Stats-iOS/InsightsTableViewController.m

```diff
@@ -699,52 +778,94 @@ - (void)configureTodaysStatsCell:(InsightsTodaysStatsTableViewCell *)cell
+        [cell.todayViewsValueButton setTitle:latestPostSummary.views forState:UIControlStateNormal];
+        [cell.todayViewsValueButton setTitleColor:latestPostSummary.viewsValue.integerValue == 0 ? [WPStyleGuide grey] : [WPStyleGuide wordPressBlue] forState:UIControlStateNormal];
+        [cell.todayLikesValueButton setTitle:latestPostSummary.likes forState:UIControlStateNormal];
+        [cell.todayLikesValueButton setTitleColor:latestPostSummary.likesValue.integerValue == 0 ? [WPStyleGuide grey] : [WPStyleGuide wordPressBlue] forState:UIControlStateNormal];
+        [cell.todayCommentsValueButton setTitle:latestPostSummary.comments forState:UIControlStateNormal];
+        [cell.todayCommentsValueButton setTitleColor:latestPostSummary.commentsValue.integerValue == 0 ? [WPStyleGuide grey] : [WPStyleGuide wordPressBlue] forState:UIControlStateNormal];
     }
 
 }
 
-- (void)configureSectionSelectableCell:(StatsSelectableTableViewCell *)cell forRow:(NSInteger)row
+- (void)configureSectionSelectableCell:(StatsSelectableTableViewCell *)cell forIndexPath:(NSIndexPath *)indexPath
 {
     cell.selectedIsLighter = NO;
     cell.unselectedCellValueColor = [WPStyleGuide wordPressBlue];
     
-    StatsSummary *todaySummary = self.sectionData[@(StatsSectionInsightsTodaysStats)];
-    
-    switch (row) {
-        case 1: // Views
-        {
-            cell.categoryIconLabel.text = @"";
-            cell.categoryLabel.text = [NSLocalizedString(@"Views", @"") uppercaseStringWithLocale:[NSLocale currentLocale]];
-            cell.valueLabel.text = todaySummary.views ?: @"-";
-            break;
-        }
-            
-        case 2: // Visitors
-        {
-            cell.categoryIconLabel.text = @"";
-            cell.categoryLabel.text = [NSLocalizedString(@"Visitors", @"") uppercaseStringWithLocale:[NSLocale currentLocale]];
-            cell.valueLabel.text = todaySummary.visitors ?: @"-";
-            break;
-        }
-            
-        case 3: // Likes
-        {
-            cell.categoryIconLabel.text = @"";
-            cell.categoryLabel.text = [NSLocalizedString(@"Likes", @"") uppercaseStringWithLocale:[NSLocale currentLocale]];
-            cell.valueLabel.text = todaySummary.likes ?: @"-";
-            break;
+    StatsSection statsSection = [self statsSectionForTableViewSection:indexPath.section];
+
+    if (statsSection == StatsSectionInsightsTodaysStats) {
+        StatsSummary *todaySummary = [self statsDataForStatsSection:statsSection];
+        
+        switch (indexPath.row) {
+            case 1: // Views
+            {
+                cell.categoryIconLabel.text = @"";
+                cell.categoryLabel.text = [NSLocalizedString(@"Views", @"") uppercaseStringWithLocale:[NSLocale currentLocale]];
+                cell.valueLabel.text = todaySummary.views ?: @"-";
+                break;
+            }
+                
+            case 2: // Visitors
```

Comment:
I am definitely doing things differently soon and moving towards subclasses with the rendering logic inside of them instead of the view controller.